qjs.c

```diff
@@ -384,20 +395,28 @@ int main(int argc, char **argv)
                 continue;
             }
             if (!strcmp(longopt, "memory-limit")) {
-                if (optind >= argc) {
-                    fprintf(stderr, "expecting memory limit");
-                    exit(1);
+                if (!opt_arg) {
+                    if (optind >= argc) {
+                        fprintf(stderr, "expecting memory limit");
+                        exit(1);
+                    }
+                    opt_arg = argv[optind++];
                 }
-                memory_limit = (size_t)strtod(argv[optind++], NULL);
-                continue;
+                // TODO(chqrlie): accept kmg suffixes
+                memory_limit = (size_t)strtod(opt_arg, NULL);
+                break;
```

Comment:
Shouldn't we continue here?

Reply:
No, we should `break` from this loop if `opt_arg` was consumed so the next argument is scanned for options.  This is somewhat tricky, I will try and find a more intuitive approach, but the logic was already there and used incorrectly in this case.